pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -204,9 +222,16 @@ type NetworkPolicyController struct {
 	// synced.
 	internalGroupQueue workqueue.RateLimitingInterface
 
-	// internalNetworkPolicyMutex protects the internalNetworkPolicyStore from
-	// concurrent access during updates to the internal NetworkPolicy object.
-	internalNetworkPolicyMutex sync.RWMutex
+	// appliedToGroupReferences tracks the reference count of policies for AppliedToGroups.
+	appliedToGroupReferences map[string]networkPolicyReferences
+	// addressGroupReferences tracks the reference count of policies for AddressGroups.
+	addressGroupReferences map[string]networkPolicyReferences
+	// appliedToGroupMutex prevents race conditions between multiple internalNetworkPolicyWorkers when they create or
+	// delete same AppliedToGroups and ensures atomicity of updating appliedToGroupStore and appliedToGroupReferences.
+	appliedToGroupMutex sync.Mutex
+	// addressGroupMutex prevents race conditions between multiple internalNetworkPolicyWorkers when they create or
+	// delete same  AddressGroups and ensures atomicity of updating addressGroupStore and addressGroupReferences.
```

Comment:
delete the same

there is also an extra whitespace

Reply:
fixed